avm/res/batch/batch-account/README.md

```diff
@@ -300,7 +300,17 @@ module batchAccount 'br/public:avm/res/batch/batch-account:<version>' = {
       {
         principalId: '<principalId>'
         principalType: 'ServicePrincipal'
-        roleDefinitionIdOrName: 'Reader'
+        roleDefinitionIdOrName: 'Owner'
+      }
+      {
+        principalId: '<principalId>'
+        principalType: 'ServicePrincipal'
+        roleDefinitionIdOrName: 'b24988ac-6180-42a0-ab88-20f7382dd24c'
+      }
+      {
+        principalId: '<principalId>'
+        principalType: 'ServicePrincipal'
+        roleDefinitionIdOrName: '<roleDefinitionIdOrName>'
```

Comment:
Wondering if this would actually clarify usage as per intention. How much effort would it be to keep the `subscriptionResourceId` function here instead? If too much I'd suggest to just open a feature request to investigate that

Reply:
Hey @eriqua, the effort was done (back in the day) to not show functions actually. Changing this might be challenging, as it may be an all or nothing (i.e., show all functions or no functions). Let's connect on this.

Reply:
For reference, this is the area where the conversion is implemented:

https://github.com/Azure/bicep-registry-modules/blob/4948e8f90b8c6e99dd39c7f8c4a37bd964de0c83/avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1#L920-L960

The specific one is 930 that checks for a function (others try to catch other cases to properly format them). 

In general, the logic we use there tries to find any
- Reference to other deployments (e.g., outputs)
- References to variables & parameters
- Functions

and replaces them with a placeholder for themselves. This was done back in the day to not show artifitial values that don't mean much to a user when looking at a usage example (e.g., let's say a reference to an output of the dependencies.bicep) as they cannot use them / don't have those same references. Further, it's a problem for when formatting the `parameter.json` examples as they can borderline not use them. 

So instead, the logic replaces those values with string representations of themselves (e.g. resourceId) instead - unless they are primitive values (e.g. booleans / numbers).

Reply:
I know the reasoning behind, but in this case didn't we add the example specifically to show how to assign a role definition which is not in the var mapping? I fear just having the <roleDefinitionIdOrName> would not meet the initial purpose. I however understand it would not be an easy one, that's why I proposed to keep it as it is and discuss the improvement in a separate issue. Would you agree to that?

Reply:
Yes and no. 2 out of 3 examples show and users can still look the 'actual' example up in the test case (as described in the header of the description). It's not perfect, but is a good reference regarless in my opinion.

Reply:
Addressed offline